app/src/main/java/com/android/unio/model/event/EventListViewModel.kt

```diff
@@ -46,11 +50,29 @@ class EventListViewModel(private val repository: EventRepository) : ViewModel()
           },
           onFailure = { exception ->
             // Handle error (e.g., log it, show a message to the user)
+            Log.e("EventViewModel", "An error occured while loading events :$exception")
             _events.value = emptyList() // Clear events on failure or handle accordingly
           })
     }
   }
 
+  fun addEvent(
+      inputStream: InputStream,
+      event: Event,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    imageRepository.uploadImage(
+        inputStream,
+        "images/events/${event.uid}",
```

Comment:
Hardcoded URL. Same comment as in "AssociationViewModel"

Reply:
Yes I think that a small task I will create a file for string constants that aren't UI-related.